packages/repository-json-schema/src/build-schema.ts

```diff
@@ -16,10 +16,44 @@ import {JSONSchema6 as JSONSchema} from 'json-schema';
 import {JSON_SCHEMA_KEY, MODEL_TYPE_KEYS} from './keys';
 
 export interface JsonSchemaOptions {
+  /**
+   * Set this flag if you want the schema to define navigational properties
+   * for model relations.
+   */
   includeRelations?: boolean;
+
+  /**
+   * Set this flag to mark all model properties as optional. This is typically
+   * used to describe request body of PATCH endpoints.
+   */
+  partial?: boolean;
+
+  /**
+   * @private
+   */
   visited?: {[key: string]: JSONSchema};
 }
 
+/**
+ * @private
+ */
+export function buildModelCacheKey(
+  ctor: Function,
+  options: JsonSchemaOptions = {},
+): string {
+  // backwards compatibility
+  switch (Object.keys(options).length) {
+    case 0:
+      return MODEL_TYPE_KEYS.ModelOnly;
+    case 1:
+      if (options.includeRelations) return MODEL_TYPE_KEYS.ModelWithRelations;
+  }
+
+  const flags = Object.keys(options);
```

Comment:
nitpick: can we define flags before the switch block and use it throughout?

Reply:
Do you meant to use `flags.length` instead of `Object.keys(options).length`? Is there any other place where to use `flags`?

Reply:
Yea that's right 👍